lib/iris/aux_factory.py

```diff
@@ -307,31 +308,6 @@ def _remap_with_bounds(self, dependency_dims, derived_dims):
             nd_values_by_key[key] = nd_values
         return nd_values_by_key
 
-    def _shape(self, nd_values_by_key):
```

Comment:
Effectively the same calc now exists inside `OceanSigmaZFactory._derive()`, but now it is done by making an array of all the dependency shapes + taking max size over each dim.